codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoSettings.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import java.util.Arrays;
+import java.util.Objects;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+/**
+ * Settings used by {@link GoCodegenPlugin}.
+ */
+public final class GoSettings {
+
+    private static final String SERVICE = "service";
+    private static final String MODULE_NAME = "module";
+    private static final String MODULE_DESCRIPTION = "moduleDescription";
+    private static final String MODULE_VERSION = "moduleVersion";
+
+    private ShapeId service;
+    private String moduleName;
+    private String moduleVersion;
+    private String moduleDescription = "";
+
+    /**
+     * Create a settings object from a configuration object node.
+     *
+     * @param config Config object to load.
+     * @return Returns the extracted settings.
+     */
+    public static GoSettings from(ObjectNode config) {
+        GoSettings settings = new GoSettings();
+        config.warnIfAdditionalProperties(Arrays.asList(SERVICE, MODULE_NAME, MODULE_DESCRIPTION, MODULE_VERSION));
+
+        settings.setService(config.expectStringMember(SERVICE).expectShapeId());
```

Comment:
Can you add a TODO for using `inferService` from Smithy's codebase instead of just fully removing it?

Reply:
To be clear, the intent is to never infer the service.